lib/graphwerk/builders/graph.rb

```diff
@@ -104,6 +104,9 @@ def add_packages_to_graph
       sig { params(package: Presenters::Package).void }
       def draw_dependencies(package)
         package.dependencies.each do |dependency|
+          unless @nodes[dependency]
+            abort "Unable to add edge `#{package.name}`->`#{dependency}`"
+          end
```

Comment:
While we're here, why not test `@nodes[package.name]` as well before we blindly assume it's present? Otherwise I  think this is a good idea.

Reply:
I don't believe it's possible for `@nodes[package.name]` to be empty because `add_packages_to_graph` iterates over `packages` (which is the source of truth) to add each one before `add_package_dependencies_to_graph` does the same thing to add the dependencies. If there were a typo in `package.name`, for example, it would be reflected in the `@nodes` keys as well.

Reply:
That may be true, but you had to look at other parts of the code in order to establish that, and future us may not notice that we've made a change that suddenly makes it possible for them to get out of sync. I don't feel strongly enough about this that I will hold up merging this because of it, but I think that in principal making that change would at least it mean that this method in isolation is "safe" from future us doing something dumb.